src/main/java/de/undercouch/gradle/tasks/download/DownloadAction.java

```diff
@@ -148,7 +148,7 @@ private void execute(URL src, Project project) throws IOException {
         }
         
         //open URL connection
-        URLConnection conn = openConnection(src, timestamp, project);
+        URLConnection conn = openConnection(src, timestamp, destFile.length(), project);
```

Comment:
Great! Anyhow, I think a quick check if the file exists will make the code more readable (just like I did with the timestamp). Something like this:

    long fileLength = 0;
    if (destFile.exists()) {
        fileLength = destFile.length();
    }

You may add this around line 122 for example.